remove multiline, remove spell name from args

TheReverendB opened the pull request:
You don't need to use multiline when returning a single command using an alias, and I would avoid passing the spell name in the args when using `!cast` as it might activate user's snippets.  I provided alternatives to this but I didn't test it (I pulled the cArgs line from an alias of mine and removed multiline and \n before {ctx.prefix} )

So, best to test it first.

Signed-off-by: TheReverendB <75177105+TheReverendB@users.noreply.github.com>

TheReverendB committed:
remove multiline, remove spell name from args

You don't need to use multiline when returning a single command using an alias, and I would avoid passing the spell name in the args when using `!cast` as it might activate user's snippets.  I provided alternatives to this but I didn't test it (I pulled the cArgs line from an alias of mine and removed multiline and \n before {ctx.prefix} )

So, best to test it first.

Signed-off-by: TheReverendB <75177105+TheReverendB@users.noreply.github.com>

ItsQc reviewed (dismissed):
Not sure yet where it's coming from but the commit as-is doesn't result in any output from Avrae. Just the ellipses showing Avrae is typing but then no message from Avrae.

ItsQc commented:
@TheReverendB 
I appreciate the suggestion 🙂 
If I can make it work I'll incorporate it. I like the idea, especially fixing the bug of spell names from being interpreted as snippets. ``multiline`` is gonna be needed every time the user successfully casts a spell by calling ``!pcast`` without including the ``i`` argument, so dropping ``multiline`` when it's not needed is just a small enhancement in my eyes.

I suspect I'll need to add a ``{ctx.prefix}multiline"``  somewhere so if a slot is used then ``multiline`` will be in order to print that trailing embed with spell slot counters. That trialing embed is needed because the ``i`` argument, which is needed to cast a spell without having it prepared according to ``!character().spellbook.prepared``, prevents the character from expending a spell slot and the resulting spell slot counters usage from showing up in the ``!cast`` embed from Avrae.

TheReverendB commented:
`character().spellbook.use_slot(2)` will use a slot, but if a player wants to use `!pcast`, they should be `!update noprep` their character to ignore prepared spells.

Make sure your <drac2> block is on the first line in your code otherwise it ends up looking to Avrae like:

!
cast

Rather than

!cast

If I have time I'll try to test it.

⁣Get BlueMail for Android ​

On Sep 9, 2022, 1:59 AM, at 1:59 AM, Qc ***@***.***> wrote:
***@***.*** 
>I appreciate the suggestion 🙂 
>However if I can make it work I'll incorporate it. I like the idea,
>especially protecting the spell names from being interpreted as
>snippets. ``multiline`` is gonna be needed in every time the user
>successfully casts a spell by calling ``!pcast`` without including the
>``i`` argument, so dropping ``multiline`` when it's not needed is lower
>priority for me.
>
>I suspect I'll need to add a ``{ctx.prefix}multiline"``  somewhere so
>if a slot is used then ``multiline`` will be in order to print that
>trailing embed with spell slot counters. That trialing embed is needed
>because the ``i`` argument, which is needed to cast a spell without
>having it prepared according to ``!character().spellbook.prepared``,
>prevents the character from expending a spell slot and the resulting
>spell slot counters usage from showing up in the ``!cast`` embed from
>Avrae.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/ItsQc/Avrae-Projects/pull/1#issuecomment-1241577031
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>

TheReverendB commented:
I just noticed!

I forgot to remove `{ctx.prefix}` 

Since this isn't a multiline anymore, it isn't needed.  That is likely the issue as `!!cast` would fail

⁣Get BlueMail for Android ​

On Sep 9, 2022, 1:59 AM, at 1:59 AM, Qc ***@***.***> wrote:
***@***.*** 
>I appreciate the suggestion 🙂 
>However if I can make it work I'll incorporate it. I like the idea,
>especially protecting the spell names from being interpreted as
>snippets. ``multiline`` is gonna be needed in every time the user
>successfully casts a spell by calling ``!pcast`` without including the
>``i`` argument, so dropping ``multiline`` when it's not needed is lower
>priority for me.
>
>I suspect I'll need to add a ``{ctx.prefix}multiline"``  somewhere so
>if a slot is used then ``multiline`` will be in order to print that
>trailing embed with spell slot counters. That trialing embed is needed
>because the ``i`` argument, which is needed to cast a spell without
>having it prepared according to ``!character().spellbook.prepared``,
>prevents the character from expending a spell slot and the resulting
>spell slot counters usage from showing up in the ``!cast`` embed from
>Avrae.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/ItsQc/Avrae-Projects/pull/1#issuecomment-1241577031
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>

ItsQc commented:
Wow. I'm gonna be completely honest, this is the first I've heard of ``!update -noprep``. I hadn't thought about the fact that update might have arguments (It's not even addressed in the help text of ``!prep``).
So..... ``!prep cast`` works again after using `` -noprep``.

Instead of scrapping this whole thing for it's redundancy (only saving 4 characters to type and adding a niche option for changing cvars), I'm gonna make the alias work both ways. And add help text about `` -noprep``. If they haven't used `` -noprep`` it'll produce the multiline workaround, otherwise the clean one that your commit will help me make.
That'll be something for the next day I have free.

TheReverendB commented:
You can follow the changelog channel and in the Avrae Development Server so it's less likely you miss updates like that.

The Development Server added built-in prepared spellcasting (and `!update noprep`) prior to the v4.1 change that broke the old version of `!prep` I repaired.

Happy aliasing! 

⁣Get BlueMail for Android ​

On Sep 9, 2022, 10:46 PM, at 10:46 PM, Qc ***@***.***> wrote:
>Wow. I'm gonna be completely honest, this is the first I've heard of
>``!update -noprep``. I had thought about the fact that update might
>have arguments (It's not even addressed in the help text of ``!prep``).
>So..... ``!prep cast`` works again after using `` -noprep``.
>
>Instead of scrapping this whole thing for it's redundancy (only saving
>4 characters to type and adding a niche option for changing cvars), I'm
>gonna make the alias work both ways. And add help text about ``
>-noprep``. If they haven't used `` -noprep`` it'll produce the
>multiline workaround, otherwise the clean one that your commit will
>help me make.
>That'll be something for the next day I have free.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/ItsQc/Avrae-Projects/pull/1#issuecomment-1242610176
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>

ItsQc committed:
Add -noprep compatibility

I discovered that ``!cast`` allows for multiple fields. Now it checks if AliasSpellbookSpell.prepared is true, if not it, instead of a second embed, it adds the spell slot counters as the last field of the spell, just like ``!cast`` does.
By this method, either a character updated with ``-noprpep`` or not will work.
Using your notes from the pull request to remove the multiline parts

Signed-off-by: Qc <89949882+ItsQc@users.noreply.github.com>

ItsQc committed:
Update pcast_docs.md

Mention ``-noprep`` and confirm compatibility with both using it not

Signed-off-by: Qc <89949882+ItsQc@users.noreply.github.com>

ItsQc committed:
Update pcast_docs.md

Add workshop entry main description.
"regardless if you used" -> "regardless of using"

Signed-off-by: Qc <89949882+ItsQc@users.noreply.github.com>

ItsQc commented:
Merged via my fork of your fork. See [my comment](https://github.com/ItsQc/Avrae-Projects/pull/2#issue-1395485943) on the separate pull request.
